### PR TITLE
Add inmemory secret store

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This change adds an inmemory version of our secret store abstraction.

I created this to help with integration-testing the new dynamic-rp component, and it felt right to send it as a separate PR to make like easier for reviewers.

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

## Contributor checklist

Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: pkg/ucp/secret/inmemory/client_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	"encoding/json"
+	"testing"
+
+	"github.com/radius-project/radius/pkg/ucp/secret"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	SecretName = "test-secret-name"
```
Comment (ytimocin) [inline pkg/ucp/secret/inmemory/client_test.go:30]: Is this exported?

Comment (rynowak) [reply pkg/ucp/secret/inmemory/client_test.go:30]: I'll fix this, and in the other packages.

**Comment by rynowak** [comment]
@ytimocin - updated

**Comment by rynowak** [comment]
Not sure what's causing the functional test issue. This change should have no impact on any actual scenario :)

**Review by lakshmimsft — APPROVED** [review]

**Review by ytimocin — APPROVED** [review]

**Pull request merged**